modules/navier_stokes/src/auxkernels/HasPorosityJumpFace.C

```diff
@@ -0,0 +1,49 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "HasPorosityJumpFace.h"
+#include "MooseMesh.h"
+#include "NS.h"
+#include "NSFVUtils.h"
+
+registerMooseObject("NavierStokesApp", HasPorosityJumpFace);
+
+InputParameters
+HasPorosityJumpFace::validParams()
+{
+  InputParameters params = AuxKernel::validParams();
+  params.addClassDescription("Shows whether an element has any attached porosity jump faces");
+  params.addRequiredParam<MooseFunctorName>(NS::porosity, "The porosity");
+  return params;
+}
+
+HasPorosityJumpFace::HasPorosityJumpFace(const InputParameters & parameters)
+  : AuxKernel(parameters), _eps(getFunctor<ADReal>(NS::porosity))
+{
+  if (isNodal())
+    mooseError("This AuxKernel only supports Elemental fields");
+}
+
+Real
+HasPorosityJumpFace::computeValue()
+{
+  for (const auto s : _current_elem->side_index_range())
+    if (const Elem * const neighbor = _current_elem->neighbor_ptr(s))
+    {
+      const FaceInfo * const fi =
```

Comment:
You could try using this:
```
Moose::FV::elemHasFaceInfo(*elem, neighbor)
```
It includes some additional checks related to AMR.

Reply:
Nice, thanks!